x-pack/legacy/plugins/reporting/server/browsers/chromium/driver/chromium_driver.ts

```diff
@@ -76,103 +80,111 @@ export class HeadlessChromiumDriver {
     logger: LevelLogger
   ): Promise<void> {
     logger.info(`opening url ${url}`);
+
     // @ts-ignore
     const client = this.page._client;
+
     let interceptedCount = 0;
 
     await this.page.setRequestInterception(true);
 
-    // We have to reach into the Chrome Devtools Protocol to apply headers as using
-    // puppeteer's API will cause map tile requests to hang indefinitely:
-    //    https://github.com/puppeteer/puppeteer/issues/5003
-    // Docs on this client/protocol can be found here:
-    //    https://chromedevtools.github.io/devtools-protocol/tot/Fetch
-    client.on('Fetch.requestPaused', async (interceptedRequest: InterceptedRequest) => {
-      const {
-        requestId,
-        request: { url: interceptedUrl },
-      } = interceptedRequest;
-      const allowed = !interceptedUrl.startsWith('file://');
-      const isData = interceptedUrl.startsWith('data:');
-
-      // We should never ever let file protocol requests go through
-      if (!allowed || !this.allowRequest(interceptedUrl)) {
-        logger.error(`Got bad URL: "${interceptedUrl}", closing browser.`);
-        await client.send('Fetch.failRequest', {
-          errorReason: 'Aborted',
+    if (!this.listenersAttached) {
```

Comment:
Since there's about 100 lines of code inside this condition, it looks like a good opportunity to bump this out to a separate class method